dcpy/utils/versions.py

```diff
@@ -306,3 +316,47 @@ def bump(
             )
         case _:
             raise ValueError(f"Unsupported version format {previous_version}")
+
+
+@dataclass(order=True)
+class DraftVersionRevision:
+    revision_num: int
+    revision_summary: str = field(compare=False)
+
+    def __post_init__(self):
+        max_string_length = 50  # random max length. may need to revise
+
+        if not isinstance(self.revision_num, int) or self.revision_num <= 0:
+            raise ValueError(
+                f"revision_num must be a positive integer, got {self.revision_num}"
+            )
+        if len(self.revision_summary) > max_string_length:
+            raise ValueError(
+                f"revision_summary must be no more than {max_string_length} characters, got {len(self.revision_summary)} characters"
+            )
+
+    @property
+    def label(self) -> str:
+        if self.revision_summary == "":
+            return f"{self.revision_num}"
+        else:
+            return f"{self.revision_num}-{self.revision_summary}"
+
+
+def parse_draft_version(v: str) -> DraftVersionRevision:
+    """Takes a version string and attempts to parse it into DraftVersionRevision object."""
+    split_index = v.find("-")
+
+    if split_index != -1:
```

Comment:
nit: if the intent is just to not "oversplit" (e.g. in the case of "1-my-dashy-draft-version", you can just add a max_split, e.g. v.split("-", 1). Then this becomes:
```
num_str, *maybe_summary = v.split("-", 1)
summary = maybe_summary[0] if len(maybe_summary) == 1 else ""
return DraftVersionRevision(int(num_str), summary)
```